work well under MacOS with Silicon Arm CPU.

luciddream-tsin opened the pull request:
- update embree to 3.13.5, and fix this: https://github.com/RenderKit/embree/commit/cda4cf191

- URHO3D_SSE OFF under MacOS Silicon.

- #include <MacTypes.h> in FreeType/src/gzip/ftzconf.h under MacOS Silicon.

eugeneko commented:
@luciddream-tsin I would like to split this PR in two, because we prefer to keep thirdparty updates in their own commits. You can do it if you like, or I can do it instead.

Also, can you elaborate how did you update embree?
The embree code is your PR is identical to this commit https://github.com/RenderKit/embree/tree/v3.13.5 plus https://github.com/RenderKit/embree/commit/cda4cf191 except for `embree\kernels\export.linux.map` and `embree\kernels\export.macosx.map`.

Were these changes intentional? Tbh I have no clue what those files do and whether we care about them.

<img width="978" height="503" alt="image" src="https://github.com/user-attachments/assets/c6507b98-054c-40a0-862b-e7fbf30c2bfa" />

luciddream-tsin commented:
Thanks for the review! Let me clarify the embree update:

I updated to embree 3.13.5 because:
- 3.13.0 was the first version to support Apple ARM chips
- 3.13.4 added 8% performance boost on Apple M1 using 8-wide BVH and double pumped NEON instructions
- The commit RenderKit/embree@cda4cf191 fixes a critical bug - two missing member variables. 

While embree 4.3.3+ versions have this fix, they would require rewriting parts of rbfx code, so I chose 3.13.5 as a safer upgrade path.

Regarding export.linux.map and export.macosx.map - I didn't intentionally modify these files. I simply copied the embree directory directly to ThirdParty. Not sure why they changed.

Please go ahead and split this PR into two parts - I'd prefer you handle it since you know the project structure better. 

Is there an alternative approach you'd recommend for the embree update?

eugeneko commented:
@luciddream-tsin I updated embree properly to the specified version: our patches to embree cmake files had to be migrated too. So, all embree-related stuff was moved out of this PR, in separate commit.

Can you verify the state of this PR?

luciddream-tsin commented:
I've compiled and tested this on Mac, and everything looks good to go, This should be ready to merge now!
